src/Core/functions.php

```diff
@@ -26,6 +26,48 @@
     define('DS', DIRECTORY_SEPARATOR);
 }
 
+if (!function_exists('Cake\Core\pathCombine')) {
+    /**
+     * Combines path segments with a forward-slash `/`ensuring
+     *  the slash is not added if it already exists.
+     *
+     * Does not sanitize the segments which may contain erroneous characters
+     * or extra and incorrect slashes.
+     *
+     * @param string ...$segments
+     * @return string
+     */
+    function pathCombine(string ...$segments): string
```

Comment:
If we're going to have options here can we used named parameters?

Reply:
Yes, planning on just adding parameters not an options array.